src/libraries/BeforeTargetFrameworkInference.targets

```diff
@@ -0,0 +1,23 @@
+<Project>
+  
+  <PropertyGroup Condition="$(TargetFramework.Contains('-'))">
+    <_OriginalTargetFramework>$(TargetFramework)</_OriginalTargetFramework>
+    <TargetFrameworkSuffix>$(TargetFramework.SubString($([MSBuild]::Add($(TargetFramework.IndexOf('-')), 1))))</TargetFrameworkSuffix>
+    <TargetFramework>$(TargetFramework.SubString(0, $(TargetFramework.IndexOf('-'))))</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <RefPath>$([MSBuild]::NormalizeDirectory('$(RefRootPath)', '$(TargetFramework)'))</RefPath>  
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisDirectory)targetframeworksuffix.props" Condition="'$(DesignTimeBuild)' == 'true'" />
```

Comment:
You are importing a props file from a targets file, that doesn't seem right. Can't you name that file `BeforeTargetFrameworkInference.props`?

Reply:
Why is this condition on DesignTimeBuild?

Reply:
The reason behind using ```BeforeTargetFrameworkInference.props``` is the we include this file after all .props and .csproj has been imported by the sdk.

```
 <Import Project="$(MSBuildThisDirectory)targetframeworksuffix.props" Condition="'$(DesignTimeBuild)' == 'true'" />
```
I kept the .props because this file is included in D.B.P for the normal build. This is still a hack because of our dependency on nuget generated files.

Reply:
> This is still a hack because of our dependency on nuget generated files.

Do we have a tracking issue to get rid of that hack, basically a feature requrest issue in https://github.com/NuGet/NuGet.Client?

Reply:
we still do not have a very firm idea about our ask here. We will require some input from the sdk team as well on how they want to support the rids in the tfm string.